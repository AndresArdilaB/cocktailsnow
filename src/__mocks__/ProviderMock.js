import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { testState } from '../frontend/initialState';
import reducer from '../frontend/reducers';

const store = createStore(reducer, testState);
const history = createBrowserHistory();

const ProviderMock = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>
      {children}
    </Router>
  </Provider>
);

export default ProviderMock;
