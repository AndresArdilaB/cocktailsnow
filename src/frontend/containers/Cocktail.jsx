import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { getCocktail } from '../utils/getApiData';
import Filters from '../components/Filters';
import CocktailDetail from '../components/CocktailDetail';

const Cocktail = ({ filters, match: { params } }) => {
  const [cocktail, setCocktail] = useState([]);
  useEffect(() => {
    getCocktail(params.id)
      .then((data) => setCocktail(data));
  }, []);

  return (
    <>
      <Filters filters={filters} isHome />
      <CocktailDetail data={{ data: cocktail[0] ? cocktail[0] : {}, type: 'Detail' }} />
    </>
  );
};

const mapStateToProps = ({ filters, currentCocktail }) => {
  return {
    filters,
    currentCocktail,
  };
};

export default connect(mapStateToProps, null)(Cocktail);
