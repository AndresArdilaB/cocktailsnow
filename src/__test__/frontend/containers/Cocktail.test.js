import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../../__mocks__/ProviderMock';
import Cocktail from '../../../frontend/containers/Cocktail';

describe('<Cocktail />', () => {

  test('Component render', () => {
    const cocktail = shallow(
      <ProviderMock>
        <Cocktail />
      </ProviderMock>,
    );

    expect(cocktail.length).toEqual(1);
  });

});

describe('Cocktail Snapshot', () => {

  test('Testing the home UI component', () => {
    const cocktail = create(
      <ProviderMock>
        <Cocktail
          match={{ params: {} }}
        />
      </ProviderMock>,
    );
    expect(cocktail.toJSON()).toMatchSnapshot();
  });
});
