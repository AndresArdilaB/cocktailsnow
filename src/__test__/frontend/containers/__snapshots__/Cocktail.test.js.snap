// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cocktail Snapshot Testing the home UI component 1`] = `
Array [
  <section
    className="filters__main"
  >
    <div
      className="filters__title"
    >
      Find your cocktail by some of this parameters
    </div>
    <section
      className="filters"
    >
      <div
        className="container"
      >
        <select
          className="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Categories
          </option>
          <option
            id="test"
            value="test"
          >
            Test
          </option>
        </select>
      </div>
      <div
        className="container"
      >
        <select
          className="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Glasses
          </option>
          <option
            id="test"
            value="test"
          >
            Test
          </option>
        </select>
      </div>
      <div
        className="container"
      >
        <select
          className="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Ingredients
          </option>
          <option
            id="test"
            value="test"
          >
            Test
          </option>
        </select>
      </div>
      <div
        className="container"
      >
        <select
          className="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Type
          </option>
          <option
            id="test"
            value="test"
          >
            Test
          </option>
        </select>
      </div>
    </section>
  </section>,
  <section
    className="detail"
  >
    <div
      className="title"
    >
      Selected Cocktel
    </div>
    <div
      className="detail-container"
    >
      <div
        className="media"
      >
        <img
          alt="Cocktail"
        />
      </div>
      <div
        className="text"
      >
        <div
          className="propierty"
        >
          <h3>
            Name:
          </h3>
          <span />
        </div>
        <div
          className="propierty"
        >
          <h3>
            Glass:
          </h3>
          <span />
        </div>
        <div
          className="propierty"
        >
          <h3>
            Category:
          </h3>
          <p />
        </div>
        <div>
          <h3>
            Ingredients:
          </h3>
        </div>
        <div>
          <h3>
            Instructions:
          </h3>
          <p />
        </div>
      </div>
    </div>
  </section>,
]
`;
